theme/helpers.js

```diff
@@ -9,13 +13,81 @@ exports.Color = class {
         Object.keys(shades).forEach((shade) => (this[shade] = shades[shade]));
     }
 
-    shade(shade) {
+    alter(shade = 500) {
 
-        return this[shade];
+        return Color(this[shade]);
     }
 
     toString() {
 
         return this['500'];
     }
 };
+
+exports.paletteFromTheme = (theme) => {
+
+    const colors = Object.keys(theme).reduce((acc, key) => {
+
+        const regex = /^color-(?<color>\w+)-(?<transparent>\w+-)?(?<shade>\d{3})/;
+
+        Assert(regex.test(key), `Expected ${key} to match regex ${regex}`);
+
+        const { color, transparent, shade } = key.match(/^color-(?<color>\w+)-(?<transparent>\w+-)?(?<shade>\d{3})/).groups;
```

Comment:
For the `transparent` group, do we need to check that the segment includes the word "transparent"? I assume for cases like `color-primary-focus`, we don't want to handle `focus` as a transparency value?

Reply:
Yea you're right!

Reply:
Reviewing this again, this helper is meant to work specifically with converting the Eva Design theme, which should always follow the regex pattern in this helper 👍